auth/auth.go

```diff
@@ -112,12 +115,28 @@ func NewClient(ctx context.Context, c *internal.AuthConfig) (*Client, error) {
 	}
 
 	return &Client{
+		hc:        &internal.HTTPClient{Client: ks.HTTPClient},
```

Comment:
Lets explicitly initialize the HTTPClient in this function, and pass it to everything that requires it.